portaudio: undefined reference to `max'

Building develop branch with mingw-w64 (for 32 bits architecture) I am getting

````
[ 47%] Linking CXX executable Release\sclang.exe
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x15bd): undefine
d reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x15e4): undefine
d reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x15fa): undefine
d reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x1763): undefine
d reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x1788): undefine
d reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x1794): more und
efined references to `max' follow
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x32d5): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x3305): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x3367): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x4f93): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x5978): un
defined reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x6283): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x651e): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x65b4): un
defined reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x6ce7): un
defined reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x95bc): un
defined reference to `min'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x987f): un
defined reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x996b): un
defined reference to `max'
../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_wdmks.c.obj):pa_win_wdmks.c:(.text+0x9b8d): un
defined reference to `max'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [lang\CMakeFiles\sclang.dir\build.make:107: lang/Release/sclang.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:3490: lang/CMakeFiles/sclang.dir/all] Error 2
````

When compiling with `-DLIBSCSYNTH=on` the linking error appears on 

````
[ 23%] Linking CXX shared library Release\liblibscsynth.dll
../../external_libraries/portaudio/portaudio_submodule/libportaudio.a(pa_win_ds.c.obj):pa_win_ds.c:(.text+0x15bd): undef
ined reference to `max'
````

It seems not portaudio fault because using

````
cmake -G"MinGW Makefiles" -DCMAKE_BUILD_TYPE=Release -DPA_BUILD_TESTS=on -DPA_BUILD_EXAMPLES=on -DPA_BUILD_SHARED=off ../supercollider\external_libraries\portaudio\portaudio_submodule
````

does compilation and linking without problems

If I add `-DPA_USE_DS=off -DPA_USE_WDMKS=off` to cmake command line, the linking problem disappears (but also DS and WDMKS disappear) ;-)

--- a/external_libraries/portaudio/CMakeLists.txt
+++ b/external_libraries/portaudio/CMakeLists.txt
@@ -13,6 +13,10 @@ set(PA_LIBNAME_ADD_SUFFIX OFF CACHE BOOL "PortAudio static library suffix" FORCE
 # disable install target
 set(PA_DISABLE_INSTALL ON CACHE BOOL "Disable PortAudio install" FORCE)
 
+if(MINGW)
+    remove_definitions(-DNOMINMAX)
+endif(MINGW)
+
 add_subdirectory(portaudio_submodule)
 
 if(WIN32 AND NOT PA_USE_ASIO)

--- a/external_libraries/portaudio/CMakeLists.txt
+++ b/external_libraries/portaudio/CMakeLists.txt
@@ -2,7 +2,7 @@
 unset(CMAKE_C_FLAGS)
 unset(CMAKE_CXX_FLAGS)
 if(WIN32)
-    remove_definitions(-DWIN32_LEAN_AND_MEAN)
+    remove_definitions(-DWIN32_LEAN_AND_MEAN -DNOMINMAX)
 endif()
 
 set(PA_BUILD_STATIC ON CACHE BOOL "PortAudio static library" FORCE)
@@ -12,10 +12,6 @@ set(PA_BUILD_SHARED OFF CACHE BOOL "PortAudio shared library" FORCE)
 set(PA_LIBNAME_ADD_SUFFIX OFF CACHE BOOL "PortAudio static library suffix" FORCE)
 # disable install target
 set(PA_DISABLE_INSTALL ON CACHE BOOL "Disable PortAudio install" FORCE)
-
-if(MINGW)
-    remove_definitions(-DNOMINMAX)
-endif(MINGW)
 
 add_subdirectory(portaudio_submodule)
 

